source/csharp/Camera/Basic/CaptureViaZivid/CaptureViaZivid.cs

```diff
@@ -0,0 +1,132 @@
+        {
+            var zivid = new Zivid.NET.Application();
+
+            Console.WriteLine("Connecting to camera");
+            var camera = zivid.ConnectCamera();
+
+            Console.WriteLine("Configuring settings");
+            var settings = new Zivid.NET.Settings
+            {
+                Acquisitions = { new Zivid.NET.Settings.Acquisition{ Aperture = 5.66,
+                                                                 ExposureTime = Duration.FromMicroseconds(8333) } },
+                Processing = { Filters = { Outlier = { Removal = { Enabled = true, Threshold = 5.0 } } } }
+            };
+
+            Console.WriteLine("Capturing frame");
+            var frame = camera.Capture(settings);
+            var pointCloud = frame.PointCloud;
+
+            Console.WriteLine("Converting to Halcon point cloud");
+            HalconDotNet.HObjectModel3D objectModel3D = ZividToHalconPointCloud(pointCloud);
+
+            string fileName = "Zivid3D.ply";
+            Console.WriteLine("Saving point cloud to: " + fileName);
+            SaveHalconPointCloud(objectModel3D, fileName);
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine("Error: " + ex.Message);
+            Environment.ExitCode = 1;
+        }
+    }
+
+    private static void SaveHalconPointCloud(HalconDotNet.HObjectModel3D model, string fileName)
+    {
+        model.WriteObjectModel3d("ply", fileName, "invert_normals", "false");
+    }
+
+    private static int FindNumberOfValidPoints(float[,,] pointCloud, ulong height, ulong width)
+    {
+        var numberOfValidPoints = 0;
+        for (ulong i = 0; i < height; i++)
+        {
+            for (ulong j = 0; j < width; j++)
+            {
+                double x = pointCloud[i, j, 0];
+                if (!Double.IsNaN(x))
+                {
+                    numberOfValidPoints = numberOfValidPoints + 1;
+                }
+            }
+        }
+        return numberOfValidPoints;
+    }
+
+    private static HalconDotNet.HObjectModel3D ZividToHalconPointCloud(Zivid.NET.PointCloud pointCloud)
+    {
+        var height = pointCloud.Height;
+        var width = pointCloud.Width;
+        var pointsXYZ = pointCloud.CopyPointsXYZ();
+        var colorsRGBA = pointCloud.CopyColorsRGBA();
+
+        // We need to find the number of valid points to initialize our halcon tuples. 
+        var numberOfValidPoints = FindNumberOfValidPoints(pointsXYZ, height, width);
+
+        // Initializing HTuples which are later filled with data from the Zivid point cloud.
+        var tupleX = new HalconDotNet.HTuple();
+        var tupleY = new HalconDotNet.HTuple();
+        var tupleZ = new HalconDotNet.HTuple();
+        var tupleR = new HalconDotNet.HTuple();
+        var tupleG = new HalconDotNet.HTuple();
+        var tupleB = new HalconDotNet.HTuple();
+        var tupleXYZMapping = new HalconDotNet.HTuple();
+
+        // Compensate for 0 indexing, therefore the ' - 1'
```

Comment:
Use the agreed convention for comments (-ing tense)

Reply:
done